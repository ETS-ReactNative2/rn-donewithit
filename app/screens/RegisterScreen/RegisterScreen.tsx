import { StyleSheet } from 'react-native'
import React from 'react'
import COLORS from '../../config/colors'
import CONSTS from './consts'
import AppSafeAreaView from '../../components/AppSafeAreaView'
import { AppForm, Field } from '../../components/forms'
import Yup from '../../utils/yup'
const fields: Field[] = [
  {
    name: 'name',
    type: 'textInput',
    defaultValue: '',
    fieldProps: {
      iconName: 'account',
      placeholder: 'Name',
      keyboardType: 'default',
      textContentType: 'name'
    }
  },
  {
    name: 'email',
    type: 'textInput',
    defaultValue: '',
    fieldProps: {
      iconName: 'email',
      placeholder: 'Email',
      keyboardType: 'email-address',
      textContentType: 'emailAddress'
    }
  },
  {
    name: 'password',
    type: 'textInput',
    defaultValue: '',
    fieldProps: {
      iconName: 'lock',
      placeholder: 'Password',
      keyboardType: 'default',
      textContentType: 'password',
      secureTextEntry: true
    }
  },
  {
    type: 'submit',
    fieldProps: {
      title: 'register'
    }
  }
]
const validationSchema = Yup.object().shape({
  name: Yup.string().required().label('Name'),
  email: Yup.string().required().email().label('Email'),
  password: Yup.string().required().min(8).label('Password')
})
export default function RegisterScreen() {
  return (
    <AppSafeAreaView style={styles.container}>
      <AppForm
        fields={fields}
        validationSchema={validationSchema}
        onSubmit={(values) => {
          console.log(values)
        }}
        style={styles.form}
      />
    </AppSafeAreaView>
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.WHITE,
    flexDirection: 'column',
    alignItems: 'center',
    paddingHorizontal: CONSTS.PADDING
  },
  form: {
    marginTop: 30
  }
})
