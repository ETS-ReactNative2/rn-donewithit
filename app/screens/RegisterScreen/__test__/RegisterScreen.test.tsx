import React from 'react'
import renderer from 'react-test-renderer'
import RegisterScreen from '../index'

describe('RegisterScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<RegisterScreen />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
