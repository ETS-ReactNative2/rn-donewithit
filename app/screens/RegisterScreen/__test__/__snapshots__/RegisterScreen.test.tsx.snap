// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "column",
        "paddingHorizontal": 10,
      },
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        Object {
          "marginTop": 30,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": undefined,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#f9f2f2",
            "borderRadius": 25,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "flex-start",
            "overflow": "hidden",
            "padding": 10,
          },
        ]
      }
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        clearButtonMode="always"
        keyboardType="default"
        maxLength={50}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Name"
        placeholderTextColor="#C0C0C0"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontFamily": "Courier",
              "fontSize": 18,
            },
            undefined,
          ]
        }
        textContentType="name"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Courier",
            "fontSize": 13,
          },
          Array [
            Object {
              "color": "tomato",
              "fontSize": 18,
              "height": 30,
              "lineHeight": 30,
              "width": "100%",
            },
            undefined,
          ],
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "width": undefined,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#f9f2f2",
            "borderRadius": 25,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "flex-start",
            "overflow": "hidden",
            "padding": 10,
          },
        ]
      }
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        clearButtonMode="always"
        keyboardType="email-address"
        maxLength={50}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="#C0C0C0"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontFamily": "Courier",
              "fontSize": 18,
            },
            undefined,
          ]
        }
        textContentType="emailAddress"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Courier",
            "fontSize": 13,
          },
          Array [
            Object {
              "color": "tomato",
              "fontSize": 18,
              "height": 30,
              "lineHeight": 30,
              "width": "100%",
            },
            undefined,
          ],
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "width": undefined,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#f9f2f2",
            "borderRadius": 25,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "flex-start",
            "overflow": "hidden",
            "padding": 10,
          },
        ]
      }
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        clearButtonMode="always"
        keyboardType="default"
        maxLength={50}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#C0C0C0"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontFamily": "Courier",
              "fontSize": 18,
            },
            undefined,
          ]
        }
        textContentType="password"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Courier",
            "fontSize": 13,
          },
          Array [
            Object {
              "color": "tomato",
              "fontSize": 18,
              "height": 30,
              "lineHeight": 30,
              "width": "100%",
            },
            undefined,
          ],
        ]
      }
    />
    <View
      style={
        Array [
          Object {},
          Array [
            Object {
              "width": "100%",
            },
            Object {
              "width": undefined,
            },
          ],
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "tomato",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Courier",
                "fontSize": 13,
              },
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
                "textTransform": "uppercase",
              },
            ]
          }
        >
          register
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
